src/javascript/app_2/App/Components/Elements/Calendar/calendar.jsx

```diff
@@ -0,0 +1,213 @@
+import moment         from 'moment';
+import PropTypes      from 'prop-types';
+import React          from 'react';
+import CalendarFooter from './calendar_footer.jsx';
+import CalendarHeader from './calendar_header.jsx';
+import CalendarPanel  from './calendar_panel.jsx';
+
+class Calendar extends React.Component {
+    constructor(props) {
+        super(props);
+        const { dateFormat, startDate } = props;
+        const current_date = (startDate ? moment(startDate) : moment()).utc().format(dateFormat);
+        this.state = {
+            calendar_date: current_date, // calendar date reference
+            selected_date: '',           // selected date
+        };
+    }
+
+    // navigates to next or previous's month/year/decade/century
+    navigators = {
+        nextMonth      : () => { this.navigateTo(1,   'months', true ); },
+        previousMonth  : () => { this.navigateTo(1,   'months', false); },
+        nextYear       : () => { this.navigateTo(1,   'years',  true ); },
+        previousYear   : () => { this.navigateTo(1,   'years',  false); },
+        nextDecade     : () => { this.navigateTo(10,  'years',  true ); },
+        previousDecade : () => { this.navigateTo(10,  'years',  false); },
+        nextCentury    : () => { this.navigateTo(100, 'years',  true ); },
+        previousCentury: () => { this.navigateTo(100, 'years',  false); },
+    }
+
+    // selects a day, a month, a year, or a decade
+    panelSelectors = {
+        date  : (e) => { this.updateSelectedDate(e); },
+        month : (e) => { this.updateSelected(e, 'month' ); },
+        year  : (e) => { this.updateSelected(e, 'year'  ); },
+        decade: (e) => { this.updateSelected(e, 'decade'); },
+    }
+
+    // sets Calendar active view
+    calendarViews = {
+        date  : () => { this.setState({ calendar_view: 'date'   }); },
+        month : () => { this.setState({ calendar_view: 'month'  }); },
+        year  : () => { this.setState({ calendar_view: 'year'   }); },
+        decade: () => { this.setState({ calendar_view: 'decade' }); },
+    }
+
+    componentWillMount() {
+        this.setState({ calendar_view: 'date' });
```

Comment:
@kellybinary 
it's better to use the constructor to set the initial value of `calendar_view`.